transport/controller.cc

```diff
@@ -65,6 +66,124 @@ future<> controller::start_server() {
+            (!cfg.native_transport_port_ssl.is_set() && !cfg.native_transport_port.is_set())) {
+        // Non-SSL port is specified || neither SSL nor non-SSL ports are specified
+        configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_transport_port()}, false });
+        _listen_addresses.push_back(configs.back().addr);
+        native_port_idx = 0;
+    }
+    if (cfg.native_shard_aware_transport_port.is_set() ||
+            (!cfg.native_shard_aware_transport_port_ssl.is_set() && !cfg.native_shard_aware_transport_port.is_set())) {
+        configs.emplace_back(listen_cfg{ socket_address{ip, cfg.native_shard_aware_transport_port()}, true });
+        _listen_addresses.push_back(configs.back().addr);
+        native_shard_aware_port_idx = native_port_idx + 1;
+    }
+
+    // main should have made sure values are clean and neatish
+    if (utils::is_true(utils::get_or_default(ceo, "enabled", "false"))) {
+        auto cred = std::make_shared<seastar::tls::credentials_builder>();
+        utils::configure_tls_creds_builder(*cred, std::move(ceo)).get();
+
+        logger.info("Enabling encrypted CQL connections between client and server");
+
+        if (cfg.native_transport_port_ssl.is_set() &&
+                (!cfg.native_transport_port.is_set() ||
+                cfg.native_transport_port_ssl() != cfg.native_transport_port())) {
+            // SSL port is specified && non-SSL port is either left out or set to a different value
+            configs.emplace_back(listen_cfg{{ip, cfg.native_transport_port_ssl()}, false, cred});
+            _listen_addresses.push_back(configs.back().addr);
+        } else if (native_port_idx >= 0) {
+            configs[native_port_idx].cred = cred;
+        }
+        if (cfg.native_shard_aware_transport_port_ssl.is_set() &&
+                (!cfg.native_shard_aware_transport_port.is_set() ||
+                cfg.native_shard_aware_transport_port_ssl() != cfg.native_shard_aware_transport_port())) {
+            configs.emplace_back(listen_cfg{{ip, cfg.native_shard_aware_transport_port_ssl()}, true, std::move(cred)});
+            _listen_addresses.push_back(configs.back().addr);
+        } else if (native_shard_aware_port_idx >= 0) {
+            configs[native_shard_aware_port_idx].cred = std::move(cred);
+        }
+    }
+
+    return parallel_for_each(configs, [&cserver, keepalive](const listen_cfg & cfg) {
+        return listen(cserver, cfg.addr, cfg.cred, cfg.is_shard_aware, keepalive, std::nullopt);
+    });
+}
+
+future<> controller::start_listening_on_maintenance_socket(sharded<cql_server>& cserver) {
+    auto unix_domain_socket_permissions =
+        file_permissions::user_read | file_permissions::user_write |
+        file_permissions::group_read | file_permissions::group_write;
+
+    auto socket = _config.maintenance_socket();
+
+    if (socket == "workdir") {
+        socket = _config.work_directory() + "/cql.m";
+    }
+
+    if (socket.length() > 107) {
```

Comment:
where this 107 magic number comes from? Better not to leave such hardcoded value, I suggest you wrap it into some variable with explanatory name.

Reply:
Now, it is calculated from the size of the structure representing the Unix domain socket address.